File: arrays/pascalTriangle.java
/*
	Given numRows, generate the first numRows of pascal's 
	triangle 
*/
public List<List<Integer>> generate(int numRows){
    List<List<Integer>> tri = new ArrayList<>();
    
    if(numRows == 0)
        return tri;
    // Add first row 
    
    // ROW 1 
    tri.add(new ArrayList<>(Arrays.asList(1)));
    
    if (numRows == 1) return tri; 
    
    // ROW 2 
    tri.add(new ArrayList<>(Arrays.asList(1,1))); 

    if (numRows == 2) return tri; 
    
    for(int i = 2; i < numRows; i++){
        List<Integer> prev_row = tri.get(i-1); 
        List<Integer> new_row = new ArrayList<>(); 
        
        new_row.add(1); 
        
        for(int j=1; j< prev_row.size(); j++){
            new_row.add(prev_row.get(j-1)+
                    prev_row.get(j)); 
        }
        
        new_row.add(1);
        
        tri.add(new_row);
    }
    return tri; 
}

File: leetcode/pascalTriangle.js
/**
 * @param {number} numRows
 * @return {number[][]}
 */
var generate = function(numRows) {
    if(numRows < 1 || !numRows) { 
        return []
    }
    else { 
        let pascal = [[1]]
        for(let i = 1; i < numRows; i++) { 
            let prevPascal = pascal[i-1];
            let newPascal = new Array(i+1); 
            for(let k = 0; k <= i; k++) { 
                if(k === 0) { 
                    newPascal[0] = 1; 
                }
                else if(k === i) { 
                    newPascal[k] = 1; 
                }
                else {
                    newPascal[k] = prevPascal[k-1] + prevPascal[k]; 
                }
            }
            pascal.push(newPascal); 
        }
        return pascal; 
    }
};

